core/modules/menu/menu.admin.inc

```diff
@@ -369,7 +369,25 @@ function menu_edit_item($form, &$form_state, $type, $item, $menu) {
     );
   }
 
+  // To support contrib modules that are expecting the old way of putting all
+  // menu items of all menus into a single list, we check for whether those
+  // contrib modules have been updated by their addition of a config setting.
+  $legacy_modules = array('content_menu', 'menu_views');
+  $use_legacy_menus = FALSE;
+  foreach ($legacy_modules as $legacy_module) {
+    $use_legacy_menus |= (module_exists($legacy_module) && !config_get($legacy_module . '.settings', 'use_updated_edit_menu_item'));
```

Comment:
I like the idea of letting modules (contrib or custom) request the new form structure via config, but the manually curated list of contrib projects in core causes some head scratching... Unfortunately, I don't have a solution for this puzzle.

Reply:
I've now made the choice a config setting for the menus module.